spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue637Test.java

```diff
@@ -0,0 +1,71 @@
+package edu.umd.cs.findbugs.detect;
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+
+import org.junit.jupiter.api.Test;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+
+/**
+ * SpotBugs should check for bad combinations of date format flags in SimpleDateFormat.
+ *
+ * @see <a href="https://github.com/spotbugs/spotbugs/issues/637">GitHub issue</a>
+ * @since 4.4.2
+ */
+public class Issue637Test extends AbstractIntegrationTest {
+    private static final String DESIRED_BUG_TYPE = "FS_BAD_DATE_FORMAT_FLAG_COMBO";
+
+    /**
+     * Expects total of 25 errors that are:
+     *  - of type defined in DESIRED_BUG_TYPE
+     *  - inside the Issue637Errors class
+     *  - was called in a specified function (parameter)
+     *  - and caught on a proper line (parameter)
+     */
+    @Test
+    public void testClassWithErrors() {
+        performAnalysis("ghIssues/Issue637Errors.class");
+        assertNumOfBug637(25);
+
+        int line;
+        for (line = 28; line <= 32; line++)
```

Comment:
Multiline `for` loops should use braces to avoid the risk of "goto fail" errors.

Reply:
Fixed by calling each assert manually, line-by-line